src/protocol/damage.rs

```diff
@@ -167,13 +183,6 @@ impl From<&BadDamageError> for [u8; 32] {
             0,
             0,
             0,
-            0,
-            0,
-            0,
-            0,
-            0,
-            0,
-            0,
```

Comment:
Did we have a bug here?

Reply:
Huh? Previously `bad_value`, `minor_opcode` and `major_opcode` were ignored. Now they are parsed. That's 7 more bytes that are being parsed, so 7 less bytes of padding.

Or am I misunderstanding you?

If you are asking if "parse, then serialize" would reproduce the original bytes: I do not think we can guarantee that for anything. Whenever there is padding, some bytes are ignored and cannot be regenerated when serialising.